### DISCUSSION TITLE
Announcing Beets-audible: a much simpler and easier workflow for audiobook management

### DISCUSSION THREAD

**Discussion by Neurrone** [seanap/Plex-Audiobook-Guide#67] [Show and tell]
I've developed a [Beets Audible plugin](https://github.com/Neurrone/beets-audible) that implements the process described in this guide. Except that it is also cross-platform and much faster and easier to use than Mp3Tag.

I'd consider it beta quality. I've used it for a few weeks now. It works for the vast majority of books that I've tried, with some limitations documented in the readme.

Hope someone finds this useful. My hope is for this to become the new gold standard of audiobook management.

**Comment by Truk-22** [comment]
Looks very interesting. I use unraid and have installed the beets docker. Would you be able to flesh out your installation and useage for docker readme for complete noobs?

Many thanks in advance

**Reply by Neurrone** [reply]
Could you escribe which step you're having trouble? I agree that the current list of steps is a bit more complex (mainly because the plugin can't be installed via pip yet).

**Reply by Truk-22** [reply]
To start, is the below folder structure within the appdata folder? Or are they paths i need to make withing the docker template?

beets
  config/
  plugins/
  scripts/
    install-deps.sh # see step 3
  docker-compose.yml # see step 2

**Reply by Neurrone** [reply]
This is the path on the host (what you run Docker on).

**Comment by seanap** [comment]
This is awesome! Finally got a chance to play with it a little bit and pretty sure this is faster than me using mp3tag w/ keyboard shortcuts.  The CLI was very nice and suits this task pretty well, I was impressed.  

~~Musicbrainz was throwing a lot of errors, is this normal? ~~ Yes for now, according to beets issue #400

Is there any way to move (or delete) the books once they were successfully matched and archived?  I tried this on about 50 books and if I had to skip a book or two it was hard to figure out which ones were skipped (maybe a log summary output?)

**Reply by MartinCa** [reply]
There is functionality for that built into beets. If you create an import section in the config you can adjust how albums/books are imported. Example I use this for my regular music import:
```
import:
    write: yes
    copy: no
    move: yes
    hardlink: no
    resume: ask
    incremental: no
    quiet_fallback: skip
    timid: no
    log: /config/beet.log
```
This will move the files, ask if an import should be resumed if interrupted, timid indicates that if the match is very good it will automatically apply it, log indicates a file where stuff like skipped albums/books are logged.
Reference: https://beets.readthedocs.io/en/stable/reference/config.html#importer-options

**Comment by SenorSmartyPants** [comment]
I finally got to trying this out, had pip install problems in my LSIO beets container.
I'm still getting MBz lookups and results. I have this in my config.yaml

```
musicbrainz:
  host: localhost:5123
```

And I see this in my verbose output

```
Tagging Jim Dale - Harry Potter and the Sorcerer's Stone
No album ID found.
Search terms: Jim Dale - Harry Potter and the Sorcerer's Stone
Album might be VA: False
Searching for MusicBrainz releases with: {'release': "harry potter and the sorcerer's stone", 'artist': 'jim dale', 'tracks': '17'}
Requesting MusicBrainz release 8179f50e-3840-42ac-a9a8-25dbdcaefb02
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_track_extract
Sending event: mb_album_extract
Sending event: albuminfo_received
Candidate: J. K. Rowling read by Jim Dale - Harry Potter and the Sorcerer’s Stone (8179f50e-3840-42ac-a9a8-25dbdcaefb02)
Computing track assignment...
...done.
Success. Distance: 0.24
```

I know changing the host is a work around, but should I be seeing this? Using LSIO beets nightly.

**Reply by seanap** [reply]
I tried to clean up the install process for when I integrate this into the guide, I made sure to add all the dependencies in the config files.  Maybe try this install and see if it goes any better? Https://github.com/seanap/beets-audible

Sending MB to localhost:5123 is supposed to prevent beets from using MB, but it looks like from your output that beets is still able to reach MB.  I think it's important to use a separate beets docker for audiobooks and for music.

**Reply by SenorSmartyPants** [reply]
For some reason pip wouldn't even run in a fresh LSIO beets container for me.

I checked out your repo earlier and used it as a base to setup my custom-cont-init.d script. I noticed requests was already installed in LSIO beets, and I use extra-files, which is also already installed.

```
#!/bin/bash

apk add --no-cache --virtual=build-dependencies --upgrade git

echo "**** installing beets-audible plugin and dependencies ****"
pip3 install markdownify natsort

rm -rf /config/plugins/audible
git clone https://github.com/Neurrone/beets-audible.git /config/plugins/audible

echo "**** cleanup ****"
apk del --purge build-dependencies
rm -rf /root/.cache /tmp/*
```

I am experiment with using one beets container. But I changing to have no default config.yaml and then I'll have one config-music.yaml and one config-audiobooks.yaml. I can then call `beet --config` with the file.

**Reply by SenorSmartyPants** [reply]
I'll have to look into the MBz config some more.

**Reply by SenorSmartyPants** [reply]
Looks like that host option doesn't work as a work around in 1.6 beets.
